src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -214,12 +214,18 @@ private async Task<bool> HeartbeatAsync(CancellationToken cancellationToken)
                 var isMasterResult = heartbeatInfo.IsMasterResult;
                 var buildInfoResult = heartbeatInfo.BuildInfoResult;
 
+                var logicalSessionTimeout = isMasterResult.LogicalSessionTimeout;
+                if (_endPoint is DnsEndPoint dnsEndPoint && dnsEndPoint.Host.Contains("shard-00-00"))
+                {
+                    logicalSessionTimeout = null;
+                }
```

Comment:
This is for testing only. It will not be pushed to master.

I'm simulating the hypothetical scenario where one mongos does not report a LogicalSessionTimeout.